### PR TITLE
#20 - Remove window* references

### PR DESCRIPTION
**Pull request by JoshuaNitschke**
WIP - untested!  Tests are passing but no clue if it will actually work - will have to test that later.

### CODE REVIEW

**Comment by JoshuaNitschke** [comment]
Okay, now we are here:

```
Error: this method must be invoked in a browser environment
    at createKindeBrowserClient (D:\GIT\****\.angular\cache\19.0.2\*****\vite\deps_ssr\kinde-angular.js:14431:11)
```

https://github.com/kinde-oss/kinde-typescript-sdk/blob/a9f596bd136ee0412d3eb3b4e3c8e1aca1473421/lib/sdk/clients/browser/index.ts#L7

**Comment by JoshuaNitschke** [comment]
Auth flow working now with SSR!

https://jam.dev/c/7430311d-5ebf-413c-bd36-88d51ff18bb8

No startup errors:

image

File: projects/kinde-angular/src/lib/kinde-angular.service.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Inject, Injectable, OnDestroy } from '@angular/core';
 import {
-  defer,
+  defer, filter,
```
Comment (luukhaijes) [inline projects/kinde-angular/src/lib/kinde-angular.service.ts]:
```suggestion
  defer,
  filter,
```

File: projects/kinde-angular/src/lib/kinde-angular.service.ts [hunk 2]
```diff
@@ -31,12 +31,14 @@ export class KindeAngularService implements OnDestroy {
...
-    @Inject(KINDE_FACTORY_TOKEN) private kindeClient: KindeClient,
-    @Inject(LOCATION_TOKEN) private location: Location,
+    @Inject(KINDE_FACTORY_TOKEN) private kindeClient: KindeClient | null,
+    @Inject(DOCUMENT) private document: Document,
+    private location: Location,
     private authState: AuthStateService
   ) {
     this.shouldHandleCallback()
       .pipe(
+        filter(x => this.kindeClient !== null),
```
Comment (luukhaijes) [inline projects/kinde-angular/src/lib/kinde-angular.service.ts]:
```suggestion
        filter(_ => this.kindeClient !== null),
```

File: projects/kinde-angular/src/lib/kinde-angular.service.ts [hunk 3]
```diff
@@ -59,10 +61,11 @@ export class KindeAngularService implements OnDestroy {
   getClaim<T>(claim: string, type?: ClaimTokenType): Observable<IClaim<T> | null> {
     // todo: make reusable function to handle isAuthenticated
     return this.isAuthenticated$.pipe(
+      filter(_ => this.kindeClient !== null),
       switchMap(isAuthenticated =>
         iif(
           () => isAuthenticated,
-          from(this.kindeClient.getClaim(claim, type))  as Observable<IClaim<T>>,
+          from(this.kindeClient!.getClaim(claim, type))  as Observable<IClaim<T>>,
```
Comment (luukhaijes) [inline projects/kinde-angular/src/lib/kinde-angular.service.ts]:
```suggestion
          from(this.kindeClient!.getClaim(claim, type)) as Observable<IClaim<T>>,
```

**Review by luukhaijes — COMMENTED** [review]
Small nitpicks, one is my own mistake. Other than, code looks good!🎉

**Comment by luukhaijes** [comment]
FYI, I will test it tomorrow.

**Comment by luukhaijes** [comment]
Are you able to push your playground changes you used to test this? I think its a good addition to move to provider syntax in the demo app.

**Comment by JoshuaNitschke** [comment]
@luukhaijes I didn't realize the purpose of the playground was to test the app!  that makes a lot of sense.

I tested in the app that I am working on building with a local NPM package pointing to my branch of your repo, so nothing to push unfortunately.  Sorry about that.

**Comment by luukhaijes** [comment]
@JoshuaNitschke, yeah its used as a starter/demo/testing and such. Your workflow sounds good too, I'll have a look into that. I tested it locally and it works as it should, approving your PR. Thanks for contributing.🎉

**Review by luukhaijes — APPROVED** [review]

**Pull request merged**